src/backend/joanie/edx_imports/utils.py

```diff
@@ -34,7 +34,8 @@ def download_and_store(filename, source_path=""):
     response = requests.get(url, stream=True, timeout=3)
 
     if response.status_code == HTTPStatus.OK:
-        return default_storage.save(filename, response.raw)
+        default_storage.save(filename, response.raw)
+        return filename
```

Comment:
You should return what `save` method return because in case of conflict the `filename` can be rename https://docs.djangoproject.com/fr/5.0/ref/files/storage/#django.core.files.storage.Storage.save

And that means that in university import task, you should not save `university.logo` but whar is return by `download_and_store` : https://github.com/openfun/joanie/blob/a49a753ab778b429f5019d8e05f7dd2ce31a4aa9/src/backend/joanie/edx_imports/tasks/universities.py#L71

```python
logo=logo_downloaded
```